.github/workflows/simple.yaml

```diff
@@ -0,0 +1,138 @@
+    steps:
+      - name: Setup R 🔧
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Hello world! 💬
+        run: cat(paste0("Hello ", Sys.getenv("NAME"), "! 👋"))
+        shell: Rscript {0}
+        env:
+          NAME: "Your name"
+
+  # Second job - by default jobs will run in parallel, unless some dependencies
+  # are introduced between them.
+  build-and-check:
+    name: Build R package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo 🛎
+        # We're using an official GitHub Action.
+        uses: actions/checkout@v4.1.1
+        if: contains(env.ENABLED_STEPS, 'checkout_repository')
+        with:
+          # Action inputs (arguments/parameters).
+          # In this case we clone the repository to the directory called the same
+          # as the repository name.
+          path: ${{ github.event.repository.name }}
+
+      - name: Setup R 🔧
+        if: contains(env.ENABLED_STEPS, 'setup_r')
+        # We're using a third-party GitHub Action.
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Setup R dependencies 🔧
+        # We're using a third-party GitHub Action.
+        uses: r-lib/actions/setup-r-dependencies@v2
+        if: contains(env.ENABLED_STEPS, 'setup_dependencies')
+        with:
+          extra-packages: any::rcmdcheck
+          working-directory: ${{ github.event.repository.name }}
+
+      # Get package name and version from DESCRIPTION so that we'll know
+      # what's the name of the tar.gz file with the built package.
+      # PKGBUILD environment variable is saved as an environment variable so this
+      # information can be accessed by subsequent steps.
+      - name: Get package name 📦
+        if: contains(env.ENABLED_STEPS, 'get_package_name')
+        run: |
+          PKGBUILD="$(Rscript -e 'cat(sprintf("%s_%s.tar.gz",(dcf <- read.dcf("DESCRIPTION"))[,"Package"], dcf[,"Version"]))')"
+          echo "PKGBUILD = $PKGBUILD"
+          echo "PKGBUILD=$PKGBUILD" >> $GITHUB_ENV
+        shell: bash
+        working-directory: ${{ github.event.repository.name }}
+
+      # Build a package stored in a directory called ${{ github.event.repository.name }}
+      # --no-manual --no-build-vignettes because this requires LaTeX.
+      - name: Build R package 🏗
+        if: contains(env.ENABLED_STEPS, 'build_r_package')
+        run: |
+          R CMD build --no-manual --no-build-vignettes ${{ github.event.repository.name }}
+          ls -l
+        shell: bash
+
+      # Check tar.gz package built in previous step.
+      # --no-manual --no-vignettes because this requires LaTeX.
+      - name: Run R CMD check 🏁
+        if: contains(env.ENABLED_STEPS, 'run_r_cmd_check')
+        run: |
+          R CMD check --no-manual --no-vignettes ${{ env.PKGBUILD }}
+        shell: bash
+        env:
+          _R_CHECK_TESTS_NLINES_: 0
+          _R_CHECK_VIGNETTES_NLINES_: 0
+
+      - name: Get tar.gz file name
+        if: contains(env.ENABLED_STEPS, 'upload_artifact')
```

Comment:
`upload_artifact` for `Get tar.gz file name"?

Reply:
I originally meant to enable both `Get tar.gz file name` and `Upload package build` steps with the same string but can be separate.

Reply:
Also, this step is mostly to show how to use a custom GH action, because we already have the tar.gz name in `PKGBUILD`.